### ISSUE TITLE
Sharing code between proton-native and react-native

### ISSUE THREAD

**Issue by dchersey** [kusti8/proton-native#83]
Hi, thank you for your efforts in creating this framework! 

I have an existing RN app that is under development and would love to add a desktop (Mac, windows, linux) target to it ... I understand I'd have to build new views but my redux actions, reducers, api, etc could all be re-used by them.  

Any tips on how to structure the code so that I can keep it to one code base?

**Comment by kusti8** [comment]
When I used react Native, I mostly used ignite cause it came with a really good structure. I can't remember exactly what it was, but basically you have a folder for state changes, folder for API, folder for stateless pure components and then a folder for state full components. IMHO that would be how I would do it. If you're doing anything that requires differing libraries, make that a separate file. 

Basically, separate the different parts of it and make it modular (same API) so that to convert, you just have to change the react Native components and props to proton native components and props.
